src/sources/host_metrics.rs

```diff
@@ -692,6 +694,22 @@ fn add_collector(collector: &str, mut metrics: Vec<Metric>) -> Vec<Metric> {
     metrics
 }
 
+#[cfg(target_os = "linux")]
+pub fn init_roots() {
```

Comment:
I wonder if the `#[cfg]` shouldn't be inside this function to reduce the complexity upstream.

Reply:
In heim these functions can _only_ be implemented for linux so I put the guard here avoid this call site breaking at runtime on windows or darwin.

Reply:
Right. I was suggesting moving the `#[cfg]` inside the function, so that the function isn't omitted on non-Linux. As it stands now, it is always referenced in `fn prepare_from_opts`, so it won't compile on other OS.

Reply:
Oh yes! My apologies, I completely misunderstood what you were saying 😅